src/services/AdvancedTradingView/advancedTradingView.selectors.ts

```diff
@@ -0,0 +1,259 @@
+import { Pair, RoutablePlatform, Token, UniswapV2RoutablePlatform } from '@swapr/sdk'
+
+import { createSelector } from '@reduxjs/toolkit'
+
+import { AppState } from '../../state'
+import { BaseAppState } from './adapters/baseAdapter/base.adapter'
+import { AllTradesAndLiquidityFromAdapters, BasePair, LiquidityTypename } from './adapters/baseAdapter/base.types'
+import { UniswapV3Pair } from './adapters/uniswapV3/uniswapV3.types'
+import { AdapterKeys, AdvancedViewTransaction } from './advancedTradingView.types'
+
+const adapterLogos: { [key in AdapterKeys]: string } = {
+  swapr: UniswapV2RoutablePlatform.SWAPR.name,
+  sushiswap: UniswapV2RoutablePlatform.SUSHISWAP.name,
+  uniswapV2: UniswapV2RoutablePlatform.UNISWAP.name,
+  honeyswap: UniswapV2RoutablePlatform.HONEYSWAP.name,
+  uniswapV3: RoutablePlatform.UNISWAP.name,
+}
+
+const formatAdapterAmount = (amount: Number) => {
+  if (amount < 0.000001) return amount.toFixed(4)
+
+  if (amount < 0.001) return amount.toFixed(6)
+
+  return amount.toFixed(4)
+}
+
+export const sortsBeforeTokens = (inputToken: Token, outputToken: Token) =>
+  inputToken.sortsBefore(outputToken) ? [inputToken, outputToken] : [outputToken, inputToken]
+
+export const getAdapterPair = <T extends BaseAppState>(key: AdapterKeys, platform: UniswapV2RoutablePlatform) =>
```

Comment:
```js
const selectCurrentUniswapV2Pair = getAdapterPair(AdapterKeys.UNISWAPV2, UniswapV2RoutablePlatform.UNISWAP)
```
 `UniswapV3` uses own selector because the interface of it is different

```js
const selectCurrentSwaprPair = getAdapterPair(AdapterKeys.SWAPR, UniswapV2RoutablePlatform.SWAPR)
const selectCurrentSushiPair = getAdapterPair(AdapterKeys.SUSHISWAP, UniswapV2RoutablePlatform.SUSHISWAP)
const selectCurrentUniswapV2Pair = getAdapterPair(AdapterKeys.UNISWAPV2, UniswapV2RoutablePlatform.UNISWAP)
const selectCurrentHoneyPair = getAdapterPair(AdapterKeys.HONEYSWAP, UniswapV2RoutablePlatform.HONEYSWAP)
const selectCurrentUniswapV3Pair = getAdapterPairUniswapV3(AdapterKeys.UNISWAPV3)
```